### PR TITLE
feat(op): nonce replay

### PR DESCRIPTION
**Pull request by emhane**
Lets transaction nonce replay error pass for blocks below bedrock.

### CODE REVIEW

File: crates/stages/src/stages/tx_lookup.rs [hunk 1]
```diff
@@ -153,17 +153,34 @@ impl<DB: Database> Stage<DB> for TransactionLookupStage {
...
-                            RawKey::<TxHash>::from_vec(hash),
-                            RawValue::<TxNumber>::from_vec(number),
-                        )?;
+                        txhash_cursor.insert(key, RawValue::<TxNumber>::from_vec(number))
+                    };
+
+                    #[cfg(feature = "optimism")]
+                    // below bedrock, transaction nonces can be replayed.
+                    //
+                    // for importing chain segment up until bedrock, end block may be bedrock block.
```
Comment (mattsse) [inline crates/stages/src/stages/tx_lookup.rs:166]: what's the impact of this?

does this skip txs? if so can we instead filter those txs out during import?

Comment (emhane) [reply crates/stages/src/stages/tx_lookup.rs:166]: filter them out based on what if not reading db, like it does here when it tries to insert?

Comment (mattsse) [reply crates/stages/src/stages/tx_lookup.rs:166]: this only concerns tx duplicates?
there shouldn't be too many of those
should be possible to identify those

Comment (emhane) [reply crates/stages/src/stages/tx_lookup.rs:166]: you mean hardcode the transactions?

Comment (emhane) [reply crates/stages/src/stages/tx_lookup.rs:166]: hardcoded will work but need to scrape logs first, blocking this pr

File: crates/stages/Cargo.toml [hunk 2]
```diff
@@ -82,6 +82,7 @@ test-utils = [
     "reth-stages-api/test-utils",
     "dep:tempfile",
 ]
+optimism = ["reth-primitives/optimism"]
```
Comment (mattsse) [inline crates/stages/Cargo.toml:85]: why do we need this now?

**Review by joshieDo — APPROVED** [review]
didn't test it, but seems fine

**Pull request merged**